api/lib/inet-grouping.js

```diff
@@ -103,3 +103,42 @@ export const mapRequestToSubnet = (req) => {
 
   return addr
 }
+
+/**
+ * @param {import('pg').Client} pgClient
+ * @param {import('node:http').IncomingMessage} req
+ * @param {string} stationId
+ * @param {string} inetGroup
+ * @param {function} recordTelemetryFn
+ */
+export const logNetworkInfo = async (pgClient, headers, stationId, inetGroup, recordTelemetryFn) => {
+  const { rows } = await pgClient.query(`
+    INSERT INTO network_info_update_history (day, station_id)
+    VALUES (now(), $1)
+    ON CONFLICT (day, station_id) DO NOTHING
+    RETURNING station_id
+  `, [stationId])
+
+  // Don't record to InfluxDB or delete the row if we didn't write the row
+  if (rows.length === 0) return
+
+  // Empty the table of every day before today
+  await pgClient.query(`
+    DELETE FROM network_info_update_history
+    WHERE day < NOW() - INTERVAL '1 day'
+  `)
```

Comment:
IIUC, this implements a locking scheme that supports multiple Node.js processes, using SQL as the shared state. Great idea! 👏🏻 

It makes me wonder if the design based on `(day, station_id)` is the most efficient one and whether it's worth optimising this part a bit more.

Alternative to consider - I am not sure how much better it is, just sharing it here for discussion.

A benefit of my proposalis that it makes it super easy to tweak the "once per day" interval in the future. We may not ever need that; I am mentioning it just for completeness.

(1) Change the table to have `station_id` and `last_updated_at` columns, using `station_id` as the PK. 

(2) Change the implementation here as follows:

```js
const { rows } = await pgClient.query(`
  UPDATE network_info_update_history 
  SET last_updated_at = NOW() 
  WHERE station_id = $1 AND last_updated_at < NOW() - INTERVAL '1 day'
  RETURNING 1 as needs_update
`, [stationId])

if (rows.length === 0) return

await pgClient.query(`
  DELETE FROM network_info_update_history
  WHERE last_updated_at < NOW() - INTERVAL '1 day'
`)
```

(3)
Create an index over `last_updated_at` or perhaps `(last_updated_at, station_id)`. I think the second form should support both queries above, but this needs to be verified using `EXPLAIN` or `EXPLAIN ANALYZE`.

Reply:
We agreed to use an in-memory structure, so this thread is no longer relevant.